src/librustc_resolve/resolve_imports.rs

```diff
@@ -763,8 +756,18 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
 
             match path_res {
                 PathResult::Module(module) => module,
-                PathResult::Indeterminate => return false,
-                PathResult::NonModule(..) | PathResult::Failed { .. } => return true,
```

Comment:
`used_imports.insert` should only be added to this arm, basically.

Reply:
The issue I saw with it is that after the first `PathResult::Failed` being emitted, _all_ the following are `PathResult::Indeterminate`. Do you know why this might be and what I should change to make it not be the case?